### DISCUSSION TITLE
SQLite3+libspatialite diverged on Windows vs Unix

### DISCUSSION THREAD

**Discussion by jdwight123** [MaxRev-Dev/gdal.netcore#126] [Q&A] [answered]
Write postgis data to the local Sqlite database using gdal, run it locally on Windows without any issues, and deploy it to a Linux Docker container with the following error: "SpatiaLite v4 DB found, but updating tables disabled because no linking against spatialite library!". How to solve this problem?

**Comment by MaxRev-Dev** [comment, marked as answer]
The error clearly says that the **sqlite3** library which is provided by VCPKG (and complied into bindings) is [not linked against](
https://github.com/microsoft/vcpkg/blob/582a4de14bef91df217f4f49624cf5b2b04bd7ca/ports/sqlite3/vcpkg.json#L20) **libspatialite**.

Windows behaviour differs and contains such extension because the binaries from GisInternals were used. On Unix systems we rely on VCPKG. For now I check only for [this driver list](https://github.com/MaxRev-Dev/gdal.netcore/blob/e6e26d6bad7700adab267c92c3861475842d8951/tests/MaxRev.Gdal.Core.Tests.XUnit/CommonTests.cs#L101) and it's not possible to track every extension.

This bindings won't support it, unless someone contributes and configures the pipeline to replace the sqlite3 library.
If you need such functionality consider building against external **sqlite3** which is linked against **libspatialite**.

**Reply by Fsystem** [reply]
Is it possible to directly replace sqlite3 with libspatialite?
see https://vcpkg.io/en/package/libspatialite

**Reply by MaxRev-Dev** [reply]
@Fsystem Replacement is not feasible as some drivers depend only on **`sqlite3`**.
I can enable **`libspatialite`** separately, but this won't affect SQL features. SQLite driver binary **must** be compiled with **`libspatialite`** sources, so API will know how to perform spatial queries, etc.
 
There are two options:
- fetch and compile sqlite3 separately from scratch (difficult)
- suggest a PR to VCPKG maintainers adding spatialite support (recommended)- https://github.com/microsoft/vcpkg. I can trigger the workflow on my side, once the feature become available in VCPKG.
